File: YacBoard/BoardInfo.go
package YacBoard

type BoardInfo uint32

const (
	BoardInfoNone           BoardInfo = 0
	EnPassantNone           BoardInfo = 0xff
	EnPassantMask           BoardInfo = 0xff
	EnPassantBlackA6        BoardInfo = 16
	EnPassantBlackB6        BoardInfo = 17
	EnPassantBlackC6        BoardInfo = 18
	EnPassantBlackD6        BoardInfo = 19
	EnPassantBlackE6        BoardInfo = 20
	EnPassantBlackF6        BoardInfo = 21
	EnPassantBlackG6        BoardInfo = 22
	EnPassantBlackH6        BoardInfo = 23
	EnPassantWhiteA3        BoardInfo = 40
	EnPassantWhiteB3        BoardInfo = 41
	EnPassantWhiteC3        BoardInfo = 42
	EnPassantWhiteD3        BoardInfo = 43
	EnPassantWhiteE3        BoardInfo = 44
	EnPassantWhiteF3        BoardInfo = 45
	EnPassantWhiteG3        BoardInfo = 46
	EnPassantWhiteH3        BoardInfo = 47
	WhiteCanCastleKingside  BoardInfo = 0x0100
	WhiteCanCastleQueenside BoardInfo = 0x0200
	BlackCanCastleKingside  BoardInfo = 0x0400
	BlackCanCastleQueenside BoardInfo = 0x0800
	HalfmoveCounterMask     BoardInfo = 0xffff0000
)

File: dart/yacboard/lib/yacboard/board_info.dart
// ignore_for_file: constant_identifier_names

import 'yac_board.dart';

extension BoardInfo on YacBoard {
  static const BoardInfoNone = 0;
  static const EnPassantNone = 0xff;
  static const EnPassantMask = 0xff;
  static const EnPassantBlackA6 = 16;
  static const EnPassantBlackB6 = 17;
  static const EnPassantBlackC6 = 18;
  static const EnPassantBlackD6 = 19;
  static const EnPassantBlackE6 = 20;
  static const EnPassantBlackF6 = 21;
  static const EnPassantBlackG6 = 22;
  static const EnPassantBlackH6 = 23;
  static const EnPassantWhiteA3 = 40;
  static const EnPassantWhiteB3 = 41;
  static const EnPassantWhiteC3 = 42;
  static const EnPassantWhiteD3 = 43;
  static const EnPassantWhiteE3 = 44;
  static const EnPassantWhiteF3 = 45;
  static const EnPassantWhiteG3 = 46;
  static const EnPassantWhiteH3 = 47;
  static const WhiteCanCastleKingside = 0x0100;
  static const WhiteCanCastleQueenside = 0x0200;
  static const BlackCanCastleKingside = 0x0400;
  static const BlackCanCastleQueenside = 0x0800;
  static const HalfmoveClockMask = 0x7fff0000;

  void setBoardInfo(int boardInfo) {
    enPassantPos = boardInfo & EnPassantMask;
    whiteCanCastleKingside = (boardInfo & WhiteCanCastleKingside) != BoardInfoNone;
    whiteCanCastleQueenside = (boardInfo & WhiteCanCastleQueenside) != BoardInfoNone;
    blackCanCastleKingside = (boardInfo & BlackCanCastleKingside) != BoardInfoNone;
    blackCanCastleQueenside = (boardInfo & BlackCanCastleQueenside) != BoardInfoNone;
    halfmoveClock = (boardInfo & HalfmoveClockMask) >> 16;
  }
}
